src/test/java/net/minecraftforge/debug/ConfigTest.java

```diff
@@ -99,4 +133,48 @@ public SubCat(String value)
             }
         }
     }
+
+    public static class ConfigurationGui extends GuiConfig
```

Comment:
Honestly can we not supply a generic GUI Class implementation that does all the element collection for them? We should be able to respect the hierarchy of the class with @Config.